### ISSUE TITLE
Halve the learning rate after specific training epoch?

### ISSUE THREAD

**Issue by MrWanter** [fchollet/keras#2795]
I'm using keras recently, I want to halve the learning rate after specific training epoch, but I don't know how to do it, I only got the decay parameter in `SGD()`, can someone help me and give me a hint? Thanks in advance!

**Comment by geovedi** [comment]
`keras.callbacks.LearningRateScheduler(schedule)` can do what you want. see #888

**Comment by MrWanter** [comment]
Thank you very much! I'll have a try.

**Comment by ishank26** [comment]
A more simpler solution for decay after specified epochs.
``` python
class decay_lr(Callback):
    ''' 
        n_epoch = no. of epochs after decay should happen.
        decay = decay value
    '''  
    def __init__(self, n_epoch, decay):
        super(decay_lr, self).__init__()
        self.n_epoch=n_epoch
        self.decay=decay

    def on_epoch_begin(self, epoch, logs={}):
        old_lr = self.model.optimizer.lr.get_value()
        if epoch%self.n_epoch == 0 :
            new_lr= self.decay*old_lr
            k.set_value(self.model.optimizer.lr, new_lr)
        else:
            k.set_value(self.model.optimizer.lr, old_lr)

decaySchedule=decay_lr(10, 0.95)
```

You can use this directly without the epoch loop.
